### PR TITLE
compat with ops 2.10

### LINKED ISSUE #285: [IPU] `_handle_relation` is incorrectly relying on relation-broken
### Bug Description

On relation-broken, we set `current_urls` to `{}`:

https://github.com/canonical/traefik-k8s-operator/blob/3f0cc21a6aac645525107fd1f70097a5af5e1dd7/lib/charms/traefik_k8s/v1/ingress_per_unit.py#L737-L742

And then we emit a custom event with `self.realtion` - which is None on relation-broken:

https://github.com/canonical/traefik-k8s-operator/blob/3f0cc21a6aac645525107fd1f70097a5af5e1dd7/lib/charms/traefik_k8s/v1/ingress_per_unit.py#L761-L762

Now, with ops v2.10.0 (specifically [ops/1091](https://github.com/canonical/operator/pull/1091)), this fails:
```
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/charm.py", line 434, in snapshot
    'relation_name': self.relation.name,
AttributeError: 'NoneType' object has no attribute 'name'
```

### To Reproduce

Run the `traefik-k8s-operator/tests/unit/test_lib_per_unit_requires.py` utest with ops pinned to 2.10.0.

### Environment

ops 2.10.0.

### Relevant log output

```shell
File "/home/ubuntu/code/o11y/traefik-k8s-operator/tests/unit/test_lib_per_unit_requires.py", line 193, in test_ipu_events
    self.harness.remove_relation(self.rel_id)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/testing.py", line 892, in remove_relation
    self._emit_relation_broken(relation_name, relation_id, remote_app)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/testing.py", line 926, in _emit_relation_broken
    self._charm.on[relation_name].relation_broken.emit(relation, app)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 351, in emit
    framework._emit(event)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 853, in _emit
    self._reemit(event_path)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 943, in _reemit
    custom_handler(event)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/lib/charms/traefik_k8s/v1/ingress_per_unit.py", line 753, in _handle_relation
    self.on.revoked_for_unit.emit(self.relation)  # type: ignore
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 351, in emit
    framework._emit(event)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 848, in _emit
    self.save_snapshot(event)
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/framework.py", line 717, in save_snapshot
    data = value.snapshot()
  File "/home/ubuntu/code/o11y/traefik-k8s-operator/.tox/unit/lib/python3.10/site-packages/ops/charm.py", line 434, in snapshot
    'relation_name': self.relation.name,
AttributeError: 'NoneType' object has no attribute 'name'
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PietroPasotti**
## Issue
Fixes #285 

## Solution
Rely on ops 2.10 behaviour that breaking relations don't show up in model.relations.

## Testing Instructions
Unit/integration tests should do it.

## Release Notes
- Fixed bug introduced by https://github.com/canonical/operator/pull/1091

### CODE REVIEW

**Review by mmkay — APPROVED** [review]
LGTM!

File: lib/charms/traefik_k8s/v1/ingress_per_unit.py
Comment (sed-i) [inline lib/charms/traefik_k8s/v1/ingress_per_unit.py:1]: Anything else to do with the [relation-broken hook](https://github.com/canonical/traefik-k8s-operator/blob/879e84f07ab1c526bfd96c50212224fc6a6d9af5/lib/charms/traefik_k8s/v1/ingress_per_unit.py#L342)?

**Review by sed-i — COMMENTED** [review]
Nice!
Much simpler than I anticipated.

File: lib/charms/traefik_k8s/v1/ingress_per_unit.py
Comment (PietroPasotti) [reply lib/charms/traefik_k8s/v1/ingress_per_unit.py:1]: What do you mean?

**Pull request merged**